packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -192,3 +216,37 @@ async function printInvalidFunctionsErrorMessage(
 		{ spaced: true },
 	);
 }
+
+/**
+ * Tries to fix potential invalid functions with a valid /index alternative from the Vercel build
+ * output.
+ *
+ * This deals with an edge case when using `basePath` and it creates an invalid function for the
+ * `/` route, but has a valid alternative at `/index`.
+ *
+ * @param collectedFunctions Collected functions from the Vercel build output.
+ */
+async function tryToFixInvalidFuncsWithValidIndexAlternative({
```

Comment:
this applies to all possible invalid funcs and not just the one at the root... I wonder if that's ok 🤔

Reply:
Hmm. Good question. I *think* it should be okay. Like, if you send a request to `/home`, that would be equivalent to sending a request to `/home/index.html` when using html files, so i think it should be okay to apply the same logic to invalid functions to try and fix them.

Reply:
...maybe :thinking:

Reply:
although I am not 100% sure let's go with this as I feel like it's unlikely it's going to cause issues 👍